src/Vault.sol

```diff
@@ -309,97 +442,112 @@ contract Vault is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, Reen
-            revert InvalidAsset(asset_);
+            revert DuplicateAsset(asset_);
         }
-        assetStorage.assets[asset_] = AssetParams({active: true, index: index, decimals: decimals_, idleBalance: 0});
+        assetStorage.assets[asset_] = AssetParams({active: true, index: index, decimals: decimals_});
         assetStorage.list.push(asset_);
-        emit NewAsset(asset_, decimals_, index);
-    }
 
-    function toggleAsset(address asset_, bool active) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        AssetStorage storage assetStorage = _getAssetStorage();
-        if (assetStorage.assets[asset_].decimals == 0) {
-            revert InvalidAsset(asset_);
-        }
-        assetStorage.assets[asset_].active = active;
-        emit ToggleAsset(asset_, active);
-    }
-
-    function addStrategy(address strategy, uint8 decimals_) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        if (strategy == address(0)) {
-            revert ZeroAddress();
-        }
-        StrategyStorage storage strategyStorage = _getStrategyStorage();
-        uint256 index = strategyStorage.list.length;
-
-        if (index > 0 && strategyStorage.strategies[strategy].index != 0) {
-            revert DuplicateStrategy();
-        }
-
-        strategyStorage.strategies[strategy] =
-            StrategyParams({active: true, index: index, decimals: decimals_, idleBalance: 0});
-        strategyStorage.list.push(strategy);
-        emit NewStrategy(strategy, index);
-    }
-
-    function toggleStrategy(address strategy, bool active) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        StrategyStorage storage strategyStorage = _getStrategyStorage();
-        if (strategyStorage.strategies[strategy].decimals == 0) {
-            revert InvalidStrategy(strategy);
-        }
-        strategyStorage.strategies[strategy].active = active;
-        emit ToggleStrategy(strategy, active);
+        emit NewAsset(asset_, decimals_, index);
     }
 
+    /**
+     * @notice Pauses or unpauses the vault.
+     * @param paused_ The new paused status.
+     */
     function pause(bool paused_) external onlyRole(DEFAULT_ADMIN_ROLE) {
         VaultStorage storage vaultStorage = _getVaultStorage();
-        if (rateProvider() == address(0)) {
-            revert RateProviderNotSet();
+        if (provider() == address(0)) {
+            revert ProviderNotSet();
         }
-        if (bufferStrategy() == address(0)) {
+        if (buffer() == address(0)) {
             revert BufferNotSet();
         }
         vaultStorage.paused = paused_;
         emit Pause(paused_);
     }
 
+    /**
+     * @notice Processes the accounting of the vault by calculating the total base balance.
+     * @dev This function iterates through the list of assets, gets their balances and rates,
+     *      and updates the total assets denominated in the base asset.
+     */
+    function processAccounting() public {
+        uint256 totalBaseBalance = address(this).balance;
```

Comment:
should we put the counting of `address(this).balance` under a configurable flag?

It doesn't apply to non-ETH and non-BNB vaults so could be forgotten in there